### PR TITLE
Prevent yellow screen opened by toast from being closed prematurely (BL-4380)

### PR DESCRIPTION
**Pull request by andrew-polk**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1611)

### CODE REVIEW

File: src/BloomExe/MiscUI/ToastNotifier.cs [hunk 1]
```diff
@@ -157,6 +157,11 @@ private void GoDownTimerTick(object sender, EventArgs e)
 
 		private void ToastNotifier_Click(object sender, EventArgs e)
 		{
+			// Certain scenarios (like clicking to bring up a yellow screen) need these lines else the yellow screen
+			// is closed by a close-this-dialog either already on the stack or about to be added by GoDownTimerTick.
+			_goDownTimer.Tick -= GoDownTimerTick; //_goDownTimer.Stop() didn't work for some reason
+			Application.Idle -= CloseThisCalledFromIdle;
+
```
Comment (hatton) [inline src/BloomExe/MiscUI/ToastNotifier.cs]: Any idea why closing the toast closes the yellow box? I agree, this is a workaround we can live with.

**Comment by andrew-polk** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/MiscUI/ToastNotifier.cs, line 164 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611#-KgGCJ5Ht9F3LNPeyw68-r1-164:-KgGNda2T22drtsAevfB:b-b8c0tq) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c228d73af89bec18ad3eac77ae281697cb150138/src/BloomExe/MiscUI/ToastNotifier.cs#L164)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Any idea why closing the toast closes the yellow box? I agree, this is a workaround we can live with.
</blockquote></details>

I assumed it was because the toast launches the yellow screen; so when the toast closes, it closes all it's children. Since I was operating on that assumption, I didn't look into it.
If that doesn't sound right, let me know, and I will investigate further.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611)*

File: src/BloomExe/MiscUI/ToastNotifier.cs [hunk 2]
```diff
@@ -157,6 +157,11 @@ private void GoDownTimerTick(object sender, EventArgs e)
 
 		private void ToastNotifier_Click(object sender, EventArgs e)
 		{
+			// Certain scenarios (like clicking to bring up a yellow screen) need these lines else the yellow screen
+			// is closed by a close-this-dialog either already on the stack or about to be added by GoDownTimerTick.
+			_goDownTimer.Tick -= GoDownTimerTick; //_goDownTimer.Stop() didn't work for some reason
```
Comment (hatton) [inline src/BloomExe/MiscUI/ToastNotifier.cs]: Pls add comment noting that we know this is at the cost of a zombie child process showing under Bloom in the Task manager.

**Comment by andrew-polk** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/MiscUI/ToastNotifier.cs, line 162 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611#-KgGCJ5Ht9F3LNPeyw68-r1-162:-KgGU7h5Hf4nN04Cyta9:bnt0ame) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c228d73af89bec18ad3eac77ae281697cb150138/src/BloomExe/MiscUI/ToastNotifier.cs#L162)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Pls add comment noting that we know this is at the cost of a zombie child process showing under Bloom in the Task manager.
</blockquote></details>

My understanding of the logic and my testing shows this does not allow the zombie process to stay around. CloseThisLater will be called at the end of this method which will close the toast appropriately.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611)*

**Comment by hatton** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/MiscUI/ToastNotifier.cs, line 162 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611#-KgGCJ5Ht9F3LNPeyw68-r1-162:-KgGt4GmRdyw8ZbxzU77:bna5pct) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c228d73af89bec18ad3eac77ae281697cb150138/src/BloomExe/MiscUI/ToastNotifier.cs#L162)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

My understanding of the logic and my testing shows this does not allow the zombie process to stay around. CloseThisLater will be called at the end of this method which will close the toast appropriately.
</blockquote></details>

ah, of course. Just had to expand a wee bit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1611)*

**Pull request merged**